IncludeToolbox/Formatter/IncludeFormatter.cs

```diff
@@ -134,7 +134,15 @@ private static List<IncludeLineInfo> SortIncludes(IList<IncludeLineInfo> lines,
             // order when rearranged by regex precedence groups.
             var includeLines = includeBatch
                 .Where(x => x.ContainsActiveInclude)
-                .OrderBy(x => x.IncludeContent);
+                .OrderBy(x => x.IncludeContent)
+                .ToList();
+
+            if (settings.RemoveDuplicates)
+            {
+                HashSet<string> uniqueIncludes = new HashSet<string>();
+                includeLines.RemoveAll(x => !x.ShouldBePreserved &&
```

Comment:
This way we wouldn't remove the duplicate in the following situation:

#include "blub" // $include-toolbox-preserve$
#include "blub" 

The first include would not be added to the HashSet so the second one would be the first.
Well, not big deal, this is a weird case anyways.

Reply:
Yeah, I did think about that case, but eventually decided that difference was probably enough to make the includes different.

There are rare include files that are designed to be included twice on purpose if you need to, but I've never used one like that so I haven't thought a great deal about what you'd want if you were using one.